docker/qgisserver/geomapfish_qgisserver/accesscontrol.py

```diff
@@ -277,7 +277,10 @@ def browse(path, node):
             # Transform ancestor names in LayerWMS instances
             layers = {}  # type: Dict[str, List[LayerWMS]]
             for layer in (
-                self.DBSession.query(LayerWMS).filter(LayerWMS.ogc_server_id == self.ogcserver.id).all()
+                session.query(LayerWMS)
+                .options(joinedload(LayerWMS.restrictionareas).subqueryload(RestrictionArea.roles))
+                .filter(LayerWMS.ogc_server_id == self.ogcserver.id)
+                .all()
```

Comment:
May use a subqueryload for restrictionareas also as we won't make a product between layers and restriction areas.